ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java

```diff
@@ -116,6 +119,39 @@ public void addTagHandler(String tagName, UiElementFactory factory, UiElementPop
         }
     }
 
+    public <T> T load(String filename, Class<T> model) {
+        UiElement container = load(filename);
+        if (container != null) {
+            return populateModel(container, model);
+        } else {
+            throw new MdxException("Failed to populate Ui file" + filename, new Exception("container is null"));
+        }
+    }
+
+    private <T extends UiElement, M> M populateModel(T container, Class<M> model) {
+        try {
+            M newInstance = model.getConstructor().newInstance();
```

Comment:
Use `Mdx.reflect.newInstance` here

Reply:
I will update this, but why 
```suggestion
            M newInstance = (M) Mdx.reflect.newInstance(model);
```

this introduces an unchecked cast and uses deprecated `Class.getInstance()`
(the latter can be updated, but the former is just a new possible breaking point)

Reply:
We can update the Class.getInstance() but the Mdx.reflect API is to allow for cross-platform bindings for games consoles where reflection isn't supported and needs to be computed ahead-of-time.